### PR TITLE
Refactor log

### PR DESCRIPTION
**Pull request by yuhan6665**
This is just an example of the code. I'm trying to:

- remove all errors.generated.go file
- define new log methods like `LogInfo(ctx context.Context, msg ...interface{})`
- also show caller method name

new log:
`[1350929471] proxy/freedom.(*Handler).Process: connection opened to tcp:127.0.0.1:46565, local endpoint 127.0.0.1:41428, remote endpoint 127.0.0.1:46565`
old log:
`[1350929471] proxy/freedom: connection opened to tcp:127.0.0.1:46565, local endpoint 127.0.0.1:41428, remote endpoint 127.0.0.1:46565`

### CODE REVIEW

**Comment by yuhan6665** [comment]
@RPRX

**Comment by Fangliding** [comment]
~~how about https://github.com/XTLS/Xray-core/issues/2521 ~~

File: common/errors/errors.go [hunk 1]
```diff
@@ -155,9 +156,15 @@ type ExportOption func(*ExportOptionHolder)
 
 // New returns a new error object with message formed from given arguments.
 func New(msg ...interface{}) *Error {
+	pc, _, _, _ := runtime.Caller(2)
+	details := runtime.FuncForPC(pc).Name()
+	if len(details) >= trim {
+		details = details[trim:]
```
[the remark this answers is not in the document]
Comment (yuhan6665) [reply common/errors/errors.go:162]: We could trim more, but I think method name could be helpful as well. Many logger specifically does that.
For runtime cost, old way use reflect too. Cost should be similar. In general I would prefer save people time vs cpu time

[the remark this answers is not in the document]
Comment (yuhan6665) [reply common/errors/errors.go:162]: Are you taking about whole use of Protobuf?

[the remark this answers is not in the document]
Comment (yuhan6665) [reply common/errors/errors.go:162]: ~~that's learning time, not wasted~~
A lot of people hate Protobuf, it's controversy around the original author of v2ray.
My take is it is better for backend API dev with built-in types and compatibility and encode efficiency. I think it is an interesting choice here, has both good and bad.
BTW my next priority going to improve on xray APIs..

**Comment by RPRX** [comment]
Planned for 1.9

**Comment by yuhan6665** [comment]
Given the likelihood of merge conflict with other work, I will merge to main later if no further comments..

**Comment by RPRX** [comment]
> Given the likelihood of merge conflict with other work, I will merge to main later if no further comments..

感觉在小版本改日志格式不太好，~~虽然我们的小版本已经......~~

**Comment by yuhan6665** [comment]
> > Given the likelihood of merge conflict with other work, I will merge to main later if no further comments..
> 
> 感觉在小版本改日志格式不太好，~虽然我们的小版本已经......~

那（如之前讨论）我可以加个字符串截取 这样就是纯 refactor？

**Pull request merged**